plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -2202,21 +2203,59 @@ def create_subflow(self, subflowName, flowAlias, realm='master', flowType='basic
                 data=json.dumps(newSubFlow),
                 timeout=self.connection_timeout,
                 validate_certs=self.validate_certs)
+            return response.headers["Location"].rsplit("/", 1)[-1]
```

Comment:
What if `Location` is not returned at all? Also do you know how ansible.module_utils.url handles casing of header names in the response headers (I don't, but I do remember it was a problem in the past - might be a loong time ago though)?

Reply:
Since Keycloak 4 (6 years old). As for the casing issue: The response is an urllib response which says (https://docs.python.org/3/library/urllib.request.html#module-urllib.response):
> Returns the headers of the response in the form of an [EmailMessage](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage) instance.

And the mime docs have do say (https://docs.python.org/3/library/email.message.html):
> The [EmailMessage](https://docs.python.org/3/library/email.message.html#email.message.EmailMessage) dictionary-like interface is indexed by the header names, which must be ASCII values. The values of the dictionary are strings with some extra methods. Headers are stored and returned in case-preserving form, but field names are matched case-insensitively.

So I think we should be save.

Reply:
I checked the docs, this also seems to be correct for Python 2.7 (with ansible-core 2.15/2.16). So it should be fine.